src/components/LHNOptionsList/LHNOptionsList.js

```diff
@@ -62,13 +127,38 @@ function LHNOptionsList({style, contentContainerStyles, data, onSelectRow, optio
      *
      * @return {Component}
      */
-    const renderItem = ({item}) => (
-        <OptionRowLHNDataWithFocus
-            reportID={item}
-            viewMode={optionMode}
-            shouldDisableFocusOptions={shouldDisableFocusOptions}
-            onSelectRow={onSelectRow}
-        />
+    const renderItem = useCallback(
+        ({item: reportID}) => {
+            const itemFullReport = reports[`${ONYXKEYS.COLLECTION.REPORT}${reportID}`];
+            const itemReportActions = reportActions[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${reportID}`];
+            const itemParentReportActions = reportActions[`${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${itemFullReport.parentReportID}`];
+            const itemPolicy = policy[`${ONYXKEYS.COLLECTION.POLICY}${itemFullReport.policyID}`];
+            const itemTransaction = `${ONYXKEYS.COLLECTION.TRANSACTION}${lodashGet(
+                itemParentReportActions,
+                [itemFullReport.parentReportActionID, 'originalMessage', 'IOUTransactionID'],
+                '',
+            )}`;
+            const itemComment = comments[`${ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT}${reportID}`] || '';
+            const participantPersonalDetailList = _.values(OptionsListUtils.getPersonalDetailsForAccountIDs(itemFullReport.participantAccountIDs, personalDetails));
+            return (
+                <OptionRowLHNDataWithFocus
```

Comment:
@kacper-mikolajczak any concerns with this approach? I recall you reworking this part lately.

Reply:
also @roryabraham I love the input, thanks for taking the time!!

Reply:
@adhorodyski For some reason did not get the GH notification, sorry for late reply!

The `withFocus` wrapper was introduced to avoid unnecessary heavy re-renders of every list item caused by change in `reportID` prop (from `withCurrentReportID` HOC). Instead of passing `reportID` directly we just derived the `isFocused` boolean value, so the list items won't be re-rendered even if `reportID` is different.

Personally, I find the wrapper solution a little bit too confusing, so if we can streamline the flow while retaining the perf gains (which are quite significant in case of an entire list), I am down for it! 🔥 

I have not made any measurments of suggested implementation above, but I think we can try to keep the perf by switching the direct call to Onyx's `personalDetails` with the `usePersonalDetails` hook (or its equivalent) which eliminates overhead of the object retrival.

Let me know if that helps! 👍 

CC: @roryabraham

Reply:
@kacper-mikolajczak I switched to `usePersonalDetails` hook to compare the results, but haven't noticed any significant change in render times. Since there is no change, I'd stick to using single pattern of loading this kind of data in such components (in this case using `withOnyx`) instead of mixing usage of HOCs with hooks